idunn/blocks/wikipedia.py

```diff
@@ -118,6 +129,64 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+class WikipediaCache:
+    _expire = None
+    _connection = None
+
+    @classmethod
+    def init_cache(cls):
+        from app import settings
+
+        redis_url = settings['WIKI_API_REDIS_URL']
+
+        if redis_url is not None:
+            cls._expire = int(settings['WIKI_CACHE_TIMEOUT']) # seconds
+            redis_db = settings['WIKI_CACHE_REDIS_DB']
+            redis_timeout = int(settings['WIKI_REDIS_TIMEOUT'])
+            redis_url = "redis://" + redis_url
+
+            try:
+                pool = ConnectionPool.from_url(redis_url, db=redis_db, socket_timeout=redis_timeout)
+            except RedisError:
+                logging.warning("No Redis instance available for caching", exc_info=True)
+                cls._connection = False
+
+            cls._connection = Redis(
+                connection_pool=pool
+            )
+        else:
+            logging.warning("No Redis URL has been set for caching", exc_info=True)
+            cls._connection = False
+
+    @classmethod
+    def cache_it(cls, key, f):
+        """
+        Takes function f and put its result in a redis cache.
+        It requires a prefix string to identify the name
+        of the function cached.
+        """
+        if WikipediaCache._connection is None:
+            WikipediaCache.init_cache()
+
+        def with_cache(*args, **kwargs):
+            """
+            If the redis is up we use the cache,
+            otherwise we bypass it
+            """
+            if cls._connection is not False:
+
+                if cls._connection.exists(key):
+                    result = json.loads(cls._connection.get(key).decode('utf-8'))
```

Comment:
Here there are 2 requests to redis: `EXISTS` and `GET`.  
A single `.get` should be enough and will return `None` if the key does not exist.